### PR TITLE
use pyright instead of mypy

### PR DESCRIPTION
**Pull request by lucabello**

### CODE REVIEW

File: tests/unit/charm/test_status_vs_config.py [hunk 1]
```diff
@@ -50,12 +51,8 @@ def test_unit_is_blocked_if_no_config_provided(self, is_leader, num_units):
 
             # AND the current unit could be either a leader or not
             self.harness.set_leader(is_leader)
-            self.harness.begin_with_initial_hooks()
-
-            # AND storage is attached
             self.harness.add_storage("content-from-git", attach=True)
-
-            self.harness.container_pebble_ready("git-sync")
+            self.harness.begin_with_initial_hooks()
```
Comment (sed-i) [inline tests/unit/charm/test_status_vs_config.py:55]: FYI @lucabello, most unit tests passed as soon as:
1. `container_pebble_ready` was removed; and
2. `add_storage` moved above `begin_with_initial_hooks`.

File: tox.ini [hunk 2]
```diff
@@ -48,28 +48,15 @@ commands =
...
-    types-setuptools
-    types-toml
-    charm,unit: {[testenv:unit]deps}
-    integration: {[testenv:integration]deps}
-commands =
-    charm: mypy {[vars]src_path} {posargs}
-    unit: mypy {[vars]tst_path}/unit {posargs}
-    integration: mypy {[vars]tst_path}/integration {posargs}
-
-[testenv:static-lib]
```
Comment (sed-i) [inline tox.ini:69]: The empty tox env was needed because the centralized CI fails if it's absent.
Is this no longer the case?

Comment (lucabello) [reply tox.ini:69]: It looks like it, yes; `tox` runs successfully for me when specifying a non-existing environment:
```
$ tox -e aaa
  aaa: OK (7.72 seconds)
  congratulations :) (7.78 seconds)
```

File: tests/unit/charm/test_status_vs_config.py [hunk 3]
```diff
@@ -158,98 +157,79 @@ def test_user_adds_units_and_relations_a_while_after_deployment_without_setting_
...
+        patcher = patch.object(COSConfigCharm, "_git_sync_version", property(lambda *_: "0.0.0"))
+        self.mock_version = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        # self.harness = Harness(COSConfigCharm)
+        # self.addCleanup(self.harness.cleanup)
+        #
+        # self.peer_rel_id = self.harness.add_relation("replicas", self.harness.model.app.name)
+        #
+        # self.harness.add_storage("content-from-git", attach=True)
```
Comment (sed-i) [inline tests/unit/charm/test_status_vs_config.py:179]: Drop this?

**Review by sed-i — APPROVED** [review]

**Pull request merged**